pkg/storageops/aws/aws.go

```diff
@@ -281,15 +297,43 @@ func (s *ec2Ops) getPrefixFromRootDeviceName(rootDeviceName string) (string, err
 // If not found it will try all the different devicePrefixes provided by AWS
 // such as /dev/sd and /dev/xvd and return the devicePath which is found
 // or return an error
-func (s *ec2Ops) getActualDevicePath(devicePath string) (string, error) {
-	letter := devicePath[len(devicePath)-1:]
-	devicePath = awsDevicePrefix + letter
-	if _, err := os.Stat(devicePath); err != nil {
-		devicePath = awsDevicePrefixWithX + letter
-		if _, err := os.Stat(devicePath); err != nil {
-			return "", err
+func (s *ec2Ops) getActualDevicePath(ipDevicePath string) (string, error) {
+	letter := ipDevicePath[len(ipDevicePath)-1:]
+	devicePath := awsDevicePrefix + letter
+	if _, err := os.Stat(devicePath); err == nil {
+		return devicePath, nil
+	}
+	devicePath = awsDevicePrefixWithX + letter
+	if _, err := os.Stat(devicePath); err == nil {
+		return devicePath, nil
+	}
+
+	devicePath = awsDevicePrefixWithH + letter
+	if _, err := os.Stat(devicePath); err == nil {
+		return devicePath, nil
+	}
+
+	// Check if the EBS volumes are exposed as NVMe drives
+	found := false
+	for _, instancePrefix := range nvmeInstanceTypes {
+		if strings.HasPrefix(s.instanceType, instancePrefix) {
+			found = true
+			break
 		}
 	}
+	if !found {
+		return "", fmt.Errorf("unable to map %v block device to an"+
+			" actual device path on the host", ipDevicePath)
+	}
+	// We name our EBS volumes /dev/xvd[f-p]. The nvme devices are serially
+	// named based on the input EBS volume device name is given.
+	// /dev/xvda -> /dev/nvme0 is reserved for root device.
+	// /dev/xvdf maps to /dev/nvme1 and so on
+	nvmeCount := letter[0] - 'f' + 1
+	devicePath = awsDevicePrefixNvme + strconv.FormatInt(int64(nvmeCount), 10) + "n1"
```

Comment:
so an example `devicePath` will be `/dev/nvme1n1` ? Can you update the comment with the example.